src/main/java/org/broadinstitute/consent/http/resources/Resource.java

```diff
@@ -42,6 +46,13 @@
     public final static String MEMBER = "Member";
     public final static String RESEARCHER = "Researcher";
 
+    // NOTE: implement more Postgres vendor codes as we encounter them
+    private static Map<String, Integer> vendorCodeStatusMap;
+    static {
+        vendorCodeStatusMap = new HashMap<>();
+        vendorCodeStatusMap.put("23505", Response.Status.CONFLICT.getStatusCode());
```

Comment:
It looks like `23505` is `PSQLState.UNIQUE_VIOLATION` ... it would be cleaner to map that enum value here instead of relying on the string always being what is defined here.
```suggestion
        vendorCodeStatusMap.put(PSQLState.UNIQUE_VIOLATION.getState(), Response.Status.CONFLICT.getStatusCode());
```